sync/src/relayer/block_transactions_verifier.rs

```diff
@@ -0,0 +1,46 @@
+use crate::relayer::compact_block::CompactBlock;
+use crate::relayer::error::{Error, Misbehavior};
+use ckb_core::transaction::Transaction;
+use ckb_protocol::{short_transaction_id, short_transaction_id_keys, ShortTransactionID};
+use fnv::FnvHashSet;
+use std::iter::FromIterator;
+
+pub struct BlockTransactionsVerifier {}
+
+impl BlockTransactionsVerifier {
+    pub(crate) fn verify(
+        block: &CompactBlock,
+        indexes: &[u32],
+        transactions: &[Transaction],
+    ) -> Result<(), Error> {
+        let indexes: FnvHashSet<u32> = FnvHashSet::from_iter(indexes.iter().cloned());
+
+        let short_ids: Vec<&ShortTransactionID> = block
+            .short_ids
+            .iter()
+            .enumerate()
+            .filter_map(|(i, x)| {
+                if indexes.contains(&(i as u32)) {
+                    Some(x)
+                } else {
+                    None
+                }
+            })
```

Comment:
1. `indexes` -> `indices`
2. `let short_ids = indices.iter()....` is better.
3. `BlockTransactionsInvalid` -> `InvalidBlockTransactions`

Reply:
1. https://grammarist.com/usage/indexes-indices/
2. sorry, I did not get it.
3. updated

Reply:
```
        let short_ids: Vec<&ShortTransactionID> = indexes
            .iter()
            .filter_map(|index| {
                block.short_ids.get(*index as usize)
            })
            .collect();
```